planner/core/collect_column_stats_usage_test.go

```diff
@@ -12,210 +12,302 @@
-			res: []string{"t1.c"},
+			sql: "select * from t order by c",
+			res: []string{"t.c"},
 		},
 		{
 			// LogicalTopN
-			sql: "select * from t1 order by a + b limit 10",
-			res: []string{"t1.a", "t1.b"},
+			sql: "select * from t order by a + b limit 10",
+			res: []string{"t.a", "t.b"},
 		},
 		{
 			// LogicalUnionAll
-			sql: "select * from ((select a, b from t1) union all (select a, c from t2)) as tmp where tmp.b > 2",
-			res: []string{"t1.b", "t2.c"},
-		},
-		{
-			// LogicalPartitionUnionAll
-			sql: "select * from t3 where a < 15 and b > 1",
-			res: []string{"t3.a", "t3.b"},
+			sql: "select * from ((select a, c from t) union all (select a, b from t2)) as tmp where tmp.c > 2",
+			res: []string{"t.c", "t2.b"},
 		},
 		{
 			// LogicalCTE
-			sql: "with cte(x, y) as (select a + 1, b from t1 where b > 1) select * from cte where x > 3",
-			res: []string{"t1.a", "t1.b"},
+			sql: "with cte(x, y) as (select a + 1, b from t where b > 1) select * from cte where x > 3",
+			res: []string{"t.a", "t.b"},
 		},
 		{
 			// LogicalCTE, LogicalCTETable
-			sql: "with recursive cte(x, y) as (select c, 1 from t1 union all select x + 1, y from cte where x < 5) select * from cte",
-			res: []string{"t1.c"},
+			sql: "with recursive cte(x, y) as (select c, 1 from t union all select x + 1, y from cte where x < 5) select * from cte",
+			res: []string{"t.c"},
 		},
 		{
 			// LogicalCTE, LogicalCTETable
-			sql: "with recursive cte(x, y) as (select 1, c from t1 union all select x + 1, y from cte where x < 5) select * from cte where y > 1",
-			res: []string{"t1.c"},
+			sql: "with recursive cte(x, y) as (select 1, c from t union all select x + 1, y from cte where x < 5) select * from cte where y > 1",
+			res: []string{"t.c"},
 		},
 		{
 			// LogicalCTE, LogicalCTETable
-			sql: "with recursive cte(x, y) as (select a, b from t1 union select x + 1, y from cte where x < 5) select * from cte",
-			res: []string{"t1.a", "t1.b"},
+			sql: "with recursive cte(x, y) as (select a, b from t union select x + 1, y from cte where x < 5) select * from cte",
+			res: []string{"t.a", "t.b"},
+		},
+		{
+			// LogicalPartitionUnionAll, static partition prune mode, use table ID rather than partition ID
+			pruneMode: "static",
+			sql:       "select * from pt1 where ptn < 20 and b > 1",
+			res:       []string{"pt1.b", "pt1.ptn"},
+		},
+		{
+			// dynamic partition prune mode, use table ID rather than partition ID
+			pruneMode: "dynamic",
+			sql:       "select * from pt1 where ptn < 20 and b > 1",
+			res:       []string{"pt1.b", "pt1.ptn"},
 		},
 	}
 
 	ctx := context.Background()
-	sctx := tk.Session()
-	is := dom.InfoSchema()
-	getColName := func(tblColID model.TableColumnID) (string, bool) {
-		tbl, ok := is.TableByID(tblColID.TableID)
-		if !ok {
-			return "", false
+	for _, tt := range tests {
+		if tt.sql == "select x.c, y.b from t as x join t2 as y on x.a = y.a group by x.c, y.b order by x.c" {
+			fmt.Println("hi")
```

Comment:
Useless?

Reply:
Yes, thinks for reminding. I have removed it.